applications/areg-portal/frontend/src/components/StyledButton.tsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { Button, ButtonProps } from "@mui/material";
+import { styled, darken } from "@mui/material/styles";
+
+interface StyledButtonProps extends ButtonProps {
+  bgPrimary?: boolean;
+}
+
+const CustomizedButton = styled(Button, {
```

Comment:
Why use a StyledButton instead of styling the desired variants at the theme level?

Reply:
The styled button strategy is convenient only if this button is really a special kind of buttons, different from all the other buttons in the application

Reply:
I used the styled component approach, as I saw there are a few more buttons with same styling across the app. Here are some examples. I didn't write it at theme level, as there are other buttons with simpler styling, such as the filter, or the table settings buttons.
<img width="198" alt="Screenshot 2022-08-23 at 17 30 25" src="https://user-images.githubusercontent.com/82731550/186199850-4d06ee4b-1dc7-492f-a9ba-a8f2a7211c35.png">
<img width="268" alt="Screenshot 2022-08-23 at 17 29 58" src="https://user-images.githubusercontent.com/82731550/186199863-44e594d1-b664-4ae1-adc7-3aa8535b1651.png">

Other styling buttons example (simpler):
<img width="296" alt="Screenshot 2022-08-23 at 17 32 35" src="https://user-images.githubusercontent.com/82731550/186200150-dadeccca-9658-40a6-a106-d52e4c37adb4.png">

Reply:
This is fine, just want to be sure that you are aware of the existence of variants: a button with/without borders are not special types but are already handled as variants. Also primary/secondary coloring are handled by the frameworks, ans do is the size (small, medium, big).
So I expect to have the majority of buttons being styled at the theme level. All the buttons in the screenshots looks manageable with the default theming hooks at a first glance.

Reply:
Oh, I see, I did't get it before. Moved the styles to the theme under the contained variant.